File: Professional Image Modifier 4/[C#]-Professional Image Modifier 4/C#/ImageFunctions/Modifications/CornerDetection/FAST.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Accord.Imaging;
using AForge;

namespace ImageFunctions.Modifications.CornerDetection
{
	class FAST
	{

		private string CurrentImage;

		public delegate void ImageCompleteHandler(List<IntPoint> Corners);
		public event ImageCompleteHandler ImageComplete;

		public FAST(string CurrentImage)
		{
			this.CurrentImage = CurrentImage;
		}

		public void GetCorners(int threshold, bool supress)
		{
			// create corners detector's instance
			FastCornersDetector fcd = new FastCornersDetector()
			{
				Suppress = supress,
				Threshold = threshold
			};
				
			// Apply the filter and return the points
			List<IntPoint> corners = fcd.ProcessImage(AForge.Imaging.Image.FromFile(this.CurrentImage));
			if (ImageComplete != null) ImageComplete(corners);
		}

	}
}

File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Modifications/CornerDetection/FAST.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Accord.Imaging
Imports AForge

Namespace ImageFunctions.Modifications.CornerDetection
	Friend Class FAST

		Private CurrentImage As String

		Public Delegate Sub ImageCompleteHandler(ByVal Corners As List(Of IntPoint))
		Public Event ImageComplete As ImageCompleteHandler

		Public Sub New(ByVal CurrentImage As String)
			Me.CurrentImage = CurrentImage
		End Sub

		Public Sub GetCorners(ByVal threshold As Integer, ByVal supress As Boolean)
			' create corners detector's instance
			Dim fcd As New FastCornersDetector() With {.Suppress = supress, .Threshold = threshold}

			' Apply the filter and return the points
			Dim corners As List(Of IntPoint) = fcd.ProcessImage(AForge.Imaging.Image.FromFile(Me.CurrentImage))
			RaiseEvent ImageComplete(corners)
		End Sub

	End Class
End Namespace
